pkg/syncutil/cmt/cmt.go

```diff
@@ -0,0 +1,106 @@
+package cmt
+
+import (
+	"context"
+	"sync"
+
+	"github.com/go-logr/logr"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/types"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/controller/config/process"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/logging"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/metrics"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/readiness"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/syncutil"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("data-replication").WithValues("metaKind", "CacheManagerTracker")
+
+type CacheManagerTracker struct {
+	lock sync.RWMutex
+
+	opa              syncutil.OpaDataClient
+	syncMetricsCache *syncutil.MetricsCache
+	tracker          *readiness.Tracker
+	processExcluder  *process.Excluder
+
+	// todo acpana -- integrate gvkaggregator
+}
+
+func NewCacheManager(opa syncutil.OpaDataClient, syncMetricsCache *syncutil.MetricsCache, tracker *readiness.Tracker, processExcluder *process.Excluder) *CacheManagerTracker {
+	return &CacheManagerTracker{
+		opa:              opa,
+		syncMetricsCache: syncMetricsCache,
+		tracker:          tracker,
+		processExcluder:  processExcluder,
+	}
+}
+
+func (c *CacheManagerTracker) AddGVKToSync(ctx context.Context, instance *unstructured.Unstructured) (*types.Responses, error) {
```

Comment:
Probably doesn't need to return types.Responses.  If we need it later, we can add it.  My hunch is that the callers don't care about what opa says.  They just want to know if things were successful or if they should handle an error.